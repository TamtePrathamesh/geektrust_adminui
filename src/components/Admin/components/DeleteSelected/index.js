import React from 'react';
import PropTypes from 'prop-types';
import { WithBoundary } from '../../../common';

import Style from './style.module.css';

const DeleteSelected = ({ deleteSelected }) => {
  return (
    <div className={Style.container}>
      <button data-testid="deleteSelected" className={Style.deleteSelected} onClick={() => deleteSelected()}>
        Delete Selected
      </button>
    </div>
  );
};

DeleteSelected.propTypes = {
  deleteSelected: PropTypes.func,
};

export default WithBoundary(React.memo(DeleteSelected));
