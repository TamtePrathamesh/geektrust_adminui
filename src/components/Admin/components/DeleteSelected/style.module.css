@import '../../../../style/colors.css';

.container {
    width: fit-content;
}

.deleteSelected {
    display: flex;
    cursor: pointer;
    color: var(--white);
    padding: 8px 16px;
    border: 0;
    transform: scale(0.9);
    transition: all ease;
    box-shadow: 0 1px 0 0 var(--grey_two);
    justify-content: center;
    align-items: center;
    border-radius: 18px;
    background-color: var(--pink);
    outline: none;
}

.deleteSelected:hover {
    transform: scale(1);
}

.deleteSelected:focus{
    outline: none;
    border: 0;
    outline-color: transparent;
    box-shadow: none;
}

