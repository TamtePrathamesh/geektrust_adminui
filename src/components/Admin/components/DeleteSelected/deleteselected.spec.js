import React from 'react';
import { screen, render } from '@testing-library/react';
import user from '@testing-library/user-event';
import DeleteSelected from '.';

describe('DeleteSelected', () => {
  it('should match snapshot', () => {
    const deleteFunc = jest.fn();
    expect(render(<DeleteSelected deleteSelected={deleteFunc} />).container).toMatchSnapshot();
  });

  it('should call function when button clicked', () => {
    const deleteFunc = jest.fn();
    render(<DeleteSelected deleteSelected={deleteFunc} />);
    expect(deleteFunc).toHaveBeenCalledTimes(0);
    user.click(screen.queryByTestId('deleteSelected'));
    expect(deleteFunc).toHaveBeenCalledTimes(1);
  });
});
