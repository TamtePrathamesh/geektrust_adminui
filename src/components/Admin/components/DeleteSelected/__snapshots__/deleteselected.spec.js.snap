// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteSelected should match snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <button
      class="deleteSelected"
      data-testid="deleteSelected"
    >
      Delete Selected
    </button>
  </div>
</div>
`;
